atmat/atphysics/Radiation/ElementRadiation.m

```diff
@@ -8,140 +8,162 @@
 % R.H. Helm, M.J. Lee, P.L. Morton and M. Sands
 % SLAC-PUB-1193, March 1973
 
-[quadon, ~]=getoption(varargin,'UseQuadrupoles', true);
-angle=atgetfieldvalues(ring,'BendingAngle');
-isdipole=isfinite(angle) & (angle~=0);
+[quadon, ~] = getoption(varargin, 'UseQuadrupoles', true);
+
+% Much faster with indices
+dipoleIdx = findcells(ring, 'BendingAngle');
+angle = atgetfieldvalues(ring, dipoleIdx, 'BendingAngle');
+isdipole = atgetcells(ring, 'BendingAngle');
+isdipole(dipoleIdx) = isfinite(angle) & (angle ~= 0);
```

Comment:
`findcells` and `atgetcells` return the same thing (`findcells` calls `atgetcells`). The only difference is that `findcells` returns integer indices while `atgetcells` return logical indices. Both can be used here, but logical indexing is more efficient, according to Mathworks. So a single call is enough:

```Matlab
isdipole = atgetcells(ring, 'BendingAngle');
angle = atgetfieldvalues(ring, isdipole, 'BendingAngle');
isdipole(isdipole) = (angle ~= 0);

if quadon
    isquadrupole = atgetcells(ring, 'Class', 'Quadrupole');
    pb = atgetfieldvalues(ring, isquadrupole, 'PolynomB', {2});
    isquadrupole(isquadrupole) = (pb ~= 0);
```

Reply:
This is even better